File: doc/rubysysadm/dlex/test_dll.rb
require 'dl/import'
require 'scanf'
dllex = DL.dlopen 'dll.dll'
init_list = dllex['init_list', 'PI']
printf "Enter the number of a list:"
num = $stdin.scanf("%d").first
r, rs = init_list.call num
until r.nil?
  r.struct! 'IP', :val, :next
  print "->#{r[:val]}"
	r = r[:next]
end
puts

File: doc/rubysysadm/dlex/test_dll.c
#include <stdio.h>	
#include "dll.h"
int main() {
	int num;
	printf("Enter the number of a list:");
	scanf("%d", &num);
	node head, curr;
	curr = NULL;
	head = init_list(num);
	curr = head;
	while(curr) {
		printf("->%d", curr->val);
		curr = curr->next;
	}
	puts("");
}
